direct-hs/src/Web/Direct/DirectRPC/Map.hs

```diff
@@ -46,9 +49,22 @@ decodeDomain (M.ObjectMap m) = do
     Just $ Domain did dname
 decodeDomain _ = Nothing
 
-fromGetTalks :: [User] -> M.Object -> [TalkRoom]
-fromGetTalks users (M.ObjectArray arr) = mapMaybe (decodeTalkRoom users) arr
-fromGetTalks _     _                   = []
+fromGetTalks :: [User] -> M.Object -> [(DomainId, [TalkRoom])]
+fromGetTalks users (M.ObjectArray arr) = foldr byDomain []
```

Comment:
Thank you for the advice.
I tried to rewrite `fromGetTalks` with `groupBy` and `sortBy` instead of recursive call. 396d0d9e7954c31b919749a6dfaca824095eabb1
Could you check this again?